### PR TITLE
Add 01-authorization

### PR DESCRIPTION
**Pull request by alexisluque**

### CODE REVIEW

File: 01-Authorization/auth0authorization/views.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    """Determines if the required scope is present in the access token
+    Args:
+        required_scope (str): The scope required to access the resource
+    """
+    def require_scope(f):
+        @wraps(f)
+        def decorated(*args, **kwargs):
+            token = get_token_auth_header(args[0])
+            unverified_claims = jwt.get_unverified_claims(token)
+            token_scopes = unverified_claims["scope"].split()
```
Comment (aaguiarz) [inline 01-Authorization/auth0authorization/views.py]: We are asking for unverified claims without validating the token. 

Shouldn't we be calling http://python-jose.readthedocs.io/en/latest/jwt/api.html#jose.jwt.decode?

File: 01-Authorization/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# Testing the API
+
+You can then try to do a GET to [http://localhost:8000/api/private](http://localhost:8000/api/private) which will
+throw an error if you don't send an access token signed with RS256 with the appropriate issuer and audience in the
+Authorization header. 
+
+You can also try to do a GET to 
+[http://localhost:8000/api/private-scoped/](http://localhost:8000/api/private-scoped) which will throw an error if
+you don't send an access token with the scope `read:agenda` signed with RS256 with the appropriate issuer and audience
+in the Authorization header.
```
Comment (aaguiarz) [inline 01-Authorization/README.md]: The scope should be read:messages

File: 01-Authorization/apiexample/settings.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    JWT_ISSUER = 'https://' + AUTH0_DOMAIN + '/'
+
+
+JWT_AUTH = {
+    'JWT_PAYLOAD_GET_USERNAME_HANDLER':
+        'auth0authorization.user.jwt_get_username_from_payload_handler',
+    'JWT_PUBLIC_KEY': PUBLIC_KEY,
+    'JWT_ALGORITHM': 'RS256',
+    'JWT_AUDIENCE': API_AUIDINCE,
+    'JWT_ISSUER': JWT_ISSUER,
```
Comment (aaguiarz) [inline 01-Authorization/apiexample/settings.py]: it says 'API_AUIDINCE' not AUDIENCE, the same above.

File: 01-Authorization/auth0authorization/views.py (hunk 1 above)
Comment (alexisluque) [reply 01-Authorization/auth0authorization/views.py]: The decorator `@api_view` checks for token validity. To ask for scopes, `@api_view` and `@requires_scope` should be used.

**Pull request merged**